### PR TITLE
misc

### PR DESCRIPTION
**Pull request by jschueller**
(moved some commits out of #2072)
- #2076 
- #2083
- #1944
- #2088
- #2089
- #2091
- HSIC graphs to reuse sample description
- Few fixes for Student

### CODE REVIEW

File: lib/src/Base/Algo/GaussKronrod.cxx
Comment (regislebrun) [inline lib/src/Base/Algo/GaussKronrod.cxx:91]: Why do you want to enforce this? There is no problem in evaluating an integral with a>=b, and it is even needed for backward integration!

Comment (jschueller) [reply lib/src/Base/Algo/GaussKronrod.cxx:91]: I was fixing a distribution bug, and thought that GaussKronrod assumed a<b:
```
const Scalar width = 0.5 * (b - a);
  const Scalar center = 0.5 * (a + b);
```

**Comment by regislebrun** [comment]
@jschueller does the continuous integration process generate something usable under Windows? I have a trainee who needs the last developments on pickle & Study...

**Comment by jschueller** [comment]
yes!

**Comment by regislebrun** [comment]
@jschueller could you recall me where I can find the Windows bundle? I had a look at the build_mingw artifacts but was unable to find anything (shame on me)

**Comment by regislebrun** [comment]
@jschueller ok so I have to wait for this PR to be merged then the next weekly build will do the job? Do you plan to add more developments in this PR? It is kinda blocking for the spatio-temporal compression my trainee is working on (but he will write his report in the mean time if there is no other option ;-))

**Comment by jschueller** [comment]
this PR is ready to review, I will stop pilling up commits

File: lib/src/Uncertainty/StatTests/VisualTest.cxx
Comment (regislebrun) [inline lib/src/Uncertainty/StatTests/VisualTest.cxx:82]: when i==size-1, we have p==1.0 and dist.computeQuantile(p)[0] will be (most of the time) much larger than sample.getMax(), resulting into a very distorted graph. @adutfoy suggested to remove the last point from the graph, which looks like a good idea, at least better than mine when I introduced this +0.5 shift!

File: python/src/FisherSnedecorFactory_doc.i.in
Comment (regislebrun) [inline python/src/FisherSnedecorFactory_doc.i.in:65]: This change triggers an error in my barnch https://github.com/openturns/openturns/pull/2068

File: python/src/openturns/PythonWrappingFunctions.hxx
Comment (regislebrun) [inline python/src/openturns/PythonWrappingFunctions.hxx:1770]: yes!!!

**Comment by regislebrun** [comment]
@jschueller Do you have another reviewer in mind? @sofianehaddad is on holidays for a while (and he deserved it) so the PR will not be merged before quite a long time...

**Review by regislebrun — APPROVED** [review]

**Comment by jschueller** [comment]
@regislebrun there's also the option to manually build it locally using the scripts at https://github.com/openturns/build

**Pull request merged**
